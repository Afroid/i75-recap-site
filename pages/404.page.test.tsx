import React from "react";
import { render, screen } from "@testing-library/react";
import Custom404 from "@/pages/404.page";

// Mock next/link so that in tests it renders a plain <a> we can query:
jest.mock("next/link", () => {
  // We need __esModule so Jest treats this as an ES default export
  return {
    __esModule: true,
    default: ({ href, children, ...rest }: { href: string; children: React.ReactNode }) => {
      return (
        <a data-testid="return-home-link" href={href} {...rest}>
          {children}
        </a>
      );
    },
  };
});

describe("Custom404", () => {
  it("renders a ‘Return Home’ link pointing to `/`", () => {
    render(<Custom404 />);

    // Grab the element by testid
    const anchor = screen.getByTestId("return-home-link");

    // Assertions
    expect(anchor).toHaveAttribute("href", "/");
    expect(anchor).toHaveTextContent("Return Home");

    // This is overkill but we'll now grab the element by role
    const link = screen.getByRole("link", { name: "Return Home" });

    // Assert it's in the document
    expect(link).toBeInTheDocument();
  });

  it("matches its snapshot", () => {
    const { asFragment } = render(<Custom404 />);
    expect(asFragment()).toMatchSnapshot();
  });
});
