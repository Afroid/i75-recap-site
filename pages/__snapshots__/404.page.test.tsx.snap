// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom404 matches its snapshot 1`] = `
<DocumentFragment>
  <main
    class="   min-h-[calc(100vh-3rem)]   min-h-screen   flex flex-col items-center justify-center   text-center   px-4   "
    data-testid="not-found-main"
  >
    <h1
      class="text-4xl font-bold mb-4"
    >
      Page Not Found
    </h1>
    <p
      class="text-lg mb-6 text-center max-w-md"
    >
      The page you're looking for doesn’t exist. You might’ve clicked a dead link or mistyped something.
    </p>
    <a
      class="text-blue-600 hover:underline text-lg"
      data-testid="return-home-link"
      href="/"
    >
      Return Home
    </a>
  </main>
</DocumentFragment>
`;
